### ISSUE TITLE
Balloon didn't show after devices sleep

### ISSUE THREAD

**Issue by JosephSanjaya** [skydoves/Balloon#184]
## Information
- Library Version: **1.3.4**
- Affected Devices: **Google Pixel 4**

## Bug description
when balloon is **showing**, if Device is **locked** or **sleep**, and then awake, **activity with balloon is resume the balloon is fading and dismissed**, and when I tried to show the balloon again it didn't do anything, **the balloon didn't shown**.

## Expected Behavior
Balloon is showing even after resuming from locked devices. 

## Steps
- Shown the balloon using _mBalloon.showAlignTop(view)_
- make the devices sleep
- when resuming devices, balloon will disappear
- tried to show the balloon again using _mBalloon.showAlignTop(view)_
- balloon didn't shown

## Implementations

```kotlin

    // Balloon Definitions

    init {
        mBalloon = createBalloon(context) {
            setLayout(R.layout.layout_baloon_password)
            setIsVisibleArrow(false)
            setWidthRatio(0.65f)
            setHeight(200)
            setDismissWhenClicked(false)
            setDismissWhenLifecycleOnPause(true)
            setDismissWhenOverlayClicked(false)
            setDismissWhenShowAgain(false)
            setDismissWhenTouchOutside(false)
            setFocusable(false)
            setCornerRadius(8f)
            setBackgroundColor(ColorUtils.getColor(R.color.very_soft_grey))
            setBalloonAnimation(BalloonAnimation.OVERSHOOT)
            setLifecycleOwner(owner)
        }.apply {
            getContentView().let { bv ->
                val rvContent = bv.findViewById<RecyclerView>(R.id.rvContent)
                val btnClose = bv.findViewById<ImageButton>(R.id.btnClose)
                btnClose.setOnClickListener { dismiss() }
                rvContent.let {
                    it.layoutManager = LinearLayoutManager(
                        context,
                        LinearLayoutManager.VERTICAL,
                        false
                    )
                    it.adapter = this@Balloon
                }
                mBalloonRecyclerView = rvContent
            }
        }
    }

    // Showing Balloon
    fun showBalloon(view: View, activity: Activity) {
        // Sometimes Balloon still shown top of keyboard
        // so it need to listen changes of keyboard height
        KeyboardUtils.registerSoftInputChangedListener(
            activity
        ) {
            if (mBalloon.isShowing && view.isFocused) {
                view.postDelayed(
                    {
                        mBalloon.dismiss()
                        if (it == 0) mBalloon.showAlignBottom(view)
                        else mBalloon.showAlignTop(view)
                    },
                    100
                )
            }
        }
        if (KeyboardUtils.isSoftInputVisible(activity)) {
            view.postDelayed(
                {
                    mBalloon.showAlignTop(view)
                },
                100
            )
        } else {
            view.postDelayed(
                {
                    mBalloon.showAlignBottom(view)
                },
                100
            )
        }
    }

    // Dismiss Balloon
    fun dismissBalloon() {
        mBalloon.dismiss()
    }

```

**Comment by skydoves** [comment]
Hi, @JosephSanjaya 
Sorry for the late reply.
The function `dismissWhenLifecycleOnPause(ture)` actually calls `onDestroy()` of the Balloon based on the lifecycle internally.
So If we want to just `dismiss()` the balloon, we should call the `dismiss()` on the override  `onPause` method in Activities or Fragments. I think the name of the `dismissWhenLifecycleOnPause` is quite be confused, I will fix this function on the next release. Thanks for your issue :)

**Comment by skydoves** [comment]
Included in the `1.3.5-SNAPSHOT`.

**Comment by JosephSanjaya** [comment]
No problem @skydoves, thank you for your response,
I see, so the problem in my definition is just 
`setDismissWhenLifecycleOnPause(true)`

Ok then, thank you!
